src/core/src/op/util/multi_subgraph_base.cpp

```diff
@@ -156,7 +156,9 @@ void ov::op::util::MultiSubGraphOp::validate_and_infer_type_body(
 
         auto body_parameter = params.at(input_description->m_body_parameter_index);
         auto input_partial_shape = input_value(index).get_partial_shape();
+        auto dtype = input_value(index).get_element_type();
```

Comment:
Could you add test in https://github.com/openvinotoolkit/openvino/blob/master/src/core/tests/type_prop/if.cpp that checks this logic?

Reply:
added